### PR TITLE
Fix mention and hashtag regexes and normalize profile biographies

### PR DESCRIPTION
**Pull request by MiguelX413**
This pull request fixes the totally incorrect regex for hashtags and also makes sure profile biographies are normalized.
The hashtag regex was totally incorrect in both the maximum length of 30 and inclusion of periods.
Profile biographies were not being normalized until now.

### CODE REVIEW

File: instaloader/structures.py [hunk 1]
```diff
@@ -394,9 +393,10 @@ def caption_mentions(self) -> List[str]:
             return []
         # This regular expression is modified from jStassen, adjusted to use Python's \w to
         # support Unicode and a word/beginning of string delimiter at the beginning to ensure
-        # that no email addresses join the list of mentions.
+        # that no email addresses join the list of mentions, and adjusted to ensure usernames
+        # only consisting of numbers are excluded.
```
Comment (aandergr) [inline instaloader/structures.py:397]: `@433` should match `433`, as that seems to be a legal username (that user exists).

Comment (aandergr) [inline instaloader/structures.py:397]: `@433` should match `433`, as that seems to be a legal username (that user exists).

**Comment by MiguelX413** [comment]
I believe this and $1541 are ready now @aandergr

**Comment by MiguelX413** [comment]
Should I squash these myself?

**Comment by aandergr** [comment]
Closing, as I prefer #1541 over this.

**Pull request closed without merging**